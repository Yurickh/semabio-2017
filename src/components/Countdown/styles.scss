@import 'ambience/typography.scss';
@import 'ambience/color.scss';
@import 'base/responsive-prop';

.count-down {
  font-weight: $weight-bold;

  display: flex;
  align-items: center;
  justify-content: center;

  cursor: pointer;
  letter-spacing: .4px;

  color: #FFF;
  background-color: $color-blood;

  @include responsive-prop(font-size, 24, 24);
  @include responsive-prop(width, 266, 266);
  @include responsive-prop(height, 52, 52);
  @include responsive-prop(margin-top, 12, 14);
  @include responsive-prop(padding-right, 26, 26);
  @include responsive-prop(padding-left, 26, 26);

  & > * {
    text-align: center;
  }

  & .separator {
    @include responsive-prop(margin-bottom, 12, 12);
    @include responsive-prop(padding-right, 6, 6);
    @include responsive-prop(padding-left, 6, 6);
  }

  & .label {
    font-weight: $weight-regular;

    letter-spacing: .32px;

    @include responsive-prop(font-size, 8, 8);
  }
}
