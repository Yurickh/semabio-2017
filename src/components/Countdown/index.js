import React, { Component } from 'react'

import './styles.css'

class Countdown extends Component {
	state = {
		days: 0,
		hours: 0,
		minutes: 0,
		seconds: 0,
	}

	refreshTime = () => {
		const now = new Date()
		const timeDiff = this.props.until - now

		const msInAnSecond = 1000
		const msInAMinute = 60 * msInAnSecond
		const msInAnHour = 60 * msInAMinute
		const msInADay = 24 * msInAnHour

		let remainingMs = timeDiff

		let days = Math.floor(remainingMs / msInADay)
		remainingMs = remainingMs - days * msInADay

		let hours = Math.floor(remainingMs / msInAnHour)
		remainingMs = remainingMs - hours * msInAnHour

		let minutes = Math.floor(remainingMs / msInAMinute)
		remainingMs = remainingMs - minutes * msInAMinute

		let seconds = Math.floor(remainingMs / msInAnSecond)

		this.setState({
			days,
			hours,
			minutes,
			seconds,
		})

		this.timeout = setTimeout(() => this.refreshTime(), 500)
	}

	componentDidMount() {
		this.timeout = setTimeout(() => this.refreshTime(), 500)
	}

	componentWillUnmount() {
		clearTimeout(this.timeout)
	}

	padZero = (num) => {
		return ('0000' + num.toString()).slice(-2)
	}

	render() {
		return (
			<div className="count-down">
				<div className="days">
					<div className="value">{ this.padZero(this.state.days)}</div>
					<div className="label">dias</div>
				</div>
				<div className="separator">:</div>
				<div className="hours">
					<div className="value">{ this.padZero(this.state.hours) }</div>
					<div className="label">horas</div>
				</div>
				<div className="separator">:</div>
				<div className="minutes">
					<div className="value">{ this.padZero(this.state.minutes) }</div>
					<div className="label">minutos</div>
				</div>
				<div className="separator">:</div>
				<div className="seconds">
					<div className="value">{ this.padZero(this.state.seconds) }</div>
					<div className="label">segundos</div>
				</div>
			</div>
		)
	}
}

export default Countdown
