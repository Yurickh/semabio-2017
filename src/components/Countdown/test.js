import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import Countdown from '../Countdown'

describe('Countdown', () => {
  it('should render correctly', () => {
    const tree = shallow(<Countdown />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
