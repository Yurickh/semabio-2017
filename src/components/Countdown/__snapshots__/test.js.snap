// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown should render correctly 1`] = `
<div
  className="count-down"
>
  <div
    className="days"
  >
    <div
      className="value"
    >
      00
    </div>
    <div
      className="label"
    >
      dias
    </div>
  </div>
  <div
    className="separator"
  >
    :
  </div>
  <div
    className="hours"
  >
    <div
      className="value"
    >
      00
    </div>
    <div
      className="label"
    >
      horas
    </div>
  </div>
  <div
    className="separator"
  >
    :
  </div>
  <div
    className="minutes"
  >
    <div
      className="value"
    >
      00
    </div>
    <div
      className="label"
    >
      minutos
    </div>
  </div>
  <div
    className="separator"
  >
    :
  </div>
  <div
    className="seconds"
  >
    <div
      className="value"
    >
      00
    </div>
    <div
      className="label"
    >
      segundos
    </div>
  </div>
</div>
`;
